TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java

```diff
@@ -283,6 +305,30 @@ public void run() {
         }
     }
 
+    @Override
+    public void onTangoConnectServiceFinish(int response, String message) {
+        if (response != TangoConnectResponse.TANGO_SUCCESS) {
+            Log.e(TAG, "Error connecting to Tango: " + response + ", message: " + message);
+            switchTangoLight(TangoStatus.SERVICE_NOT_CONNECTED);
+            displayToastMessage(R.string.tango_connect_error);
+            return;
+        }
+
+        displayToastMessage(R.string.tango_connect_success);
+    }
+
+    @Override
+    public void onTangoDisconnectServiceFinish(int response, String message) {
+        if (response != TangoConnectResponse.TANGO_SUCCESS) {
+            Log.e(TAG, "Error disconnecting from Tango: " + response + ", message: " + message);
+            switchTangoLight(TangoStatus.SERVICE_NOT_CONNECTED);
+            displayToastMessage(R.string.tango_disconnect_error);
+            return;
+        }
+
```

Comment:
I added an extra topic `/tango/status` which is published from `tango_ros_node` when (dis)connection failed or succeeded. This is required to make the Tango Status lights work when the `tango/connect` service is called directly from `ROS`.